examples/vmi-kernel-boot.yaml

```diff
@@ -0,0 +1,29 @@
+---
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-kernel-boot
+  name: vmi-kernel-boot
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+    firmware:
+      kernelBoot:
+        cmdline: console=ttyS0
+        kernelPath: kernelImage/
+        name: containerdisk
```

Comment:
I think it makes sense to inline something  like the `containerDisk` struct here. It just is not a volume. There are also fields on the containerDisk (like disk path) which have no meaning in this context. Let's create a dedicated struct.
Also let's think about the possibility that different sources may appear later on. A possible solution would be something like this:


```
kernelBoot:
   cmdline: <something>
   container:
     image: <image>
     imagePullSecret #optional
     imagePullPolicy #optional
     kernelPath: #optional
    <...>
```

If we want to provide it via hostPath (I don't think we should but just as an example) in the future, it would then for instance look like this:

```
kernelBoot:
   cmdline: <something>
   hostPath: # optional
     relativeKernelPath: #optional
    <...>
```

Note that I am also calling the subsection `container` and not `containerDisk`, to make clear that it is different.
Feel free to propose a file-structure which would be auto-discovered.

Reply:
We could also consider adding ad `cmdlinePath` where we would have a file which could provide a "default" commandline if none is extra specified.

Reply:
Regarding to the implementation, here a possible implementation flow:

1) virt-controller would in template.go check if a container is specified there and add it to the pod template
2) virt-handler would have in pkg/virt-handler/container-disk/mount.go another Mount function (something like MountKernelArtifacts and call this function in `pkg/virt-handler/vm.go` before or after `Mount` for the containerdisks is called, if a container is specified.
3) virt-launcher would like for the mounted containerDiskPath translate the VMI spec to the path of the artifacts in the converter (if we go with default paths, it would probably check which files are mounted and pass them in via the converter context).

Reply:
What if we will just reference a volume here? In this case, we will not need to use a specific struct here and it will be easy to extend the support in the future. We could limit the types that we can support right now.

Reply:
My suggestion was to use not a volume, since we see already on the containerDisk that it contains fields which are tailored to the need of exposing a disk.

Reply:
We had an offline conversation and I agree with Roman, it's best to reference the source as suggested.